### PR TITLE
PORT-9948 bug wrong action docs in the terraform

### PR DESCRIPTION
**Pull request by kodjomiles**
# Description

What - Wrong action docs in the terraform
Why - Content title was not matching with property description and some links were broken. misleading to client
How - Adding more content to description of port action schema property context and fixing broken links

## Type of change

Please leave one option from the following and delete the rest:
- [ ] Documentation (added/updated documentation)

### CODE REVIEW

File: port/scorecard/schema.go [hunk 1]
```diff
@@ -183,8 +183,8 @@ resource "port_scorecard" "readiness" {
             operator = "isNotEmpty"
           }),
           jsonencode({
-            property = "author",
-            operator : "=",
```
Comment (Tankilevitch) [inline port/scorecard/schema.go:187]: why you removed it?

Comment (Tankilevitch) [reply port/scorecard/schema.go:187]: will it still work?

File: port/action/schema.go [hunk 2]
```diff
@@ -678,7 +677,7 @@ func ObjectPropertySchema() schema.Attribute {
 			},
 		},
 		"encryption": schema.StringAttribute{
-			MarkdownDescription: "The algorithm to encrypt the property with",
+			MarkdownDescription: "The algorithm to encrypt the property with. Accepted value: `aes256-gcm`, `aes128-gcm`",
 			Optional:            true,
 			Validators: []validator.String{
 				stringvalidator.OneOf("aes256-gcm"),
```
Comment (Tankilevitch) [inline port/action/schema.go:683]: seems like only one is expected not two, please correct me if im wrong

File: port/action/schema.go [hunk 3]
```diff
@@ -499,7 +498,7 @@ func StringPropertySchema() schema.Attribute {
 			},
 		},
 		"encryption": schema.StringAttribute{
-			MarkdownDescription: "The algorithm to encrypt the property with",
+			MarkdownDescription: "The algorithm to encrypt the property with. Accepted value: `aes256-gcm`, `aes128-gcm`",
```
Comment (Tankilevitch) [inline port/action/schema.go:501]: same here

File: port/action/schema.go [hunk 4]
```diff
@@ -269,10 +269,9 @@ func ActionSchema() map[string]schema.Attribute {
 					MarkdownDescription: "Required when selecting type WEBHOOK. The URL to invoke the action",
 					Required:            true,
 				},
-				"agent": schema.StringAttribute{
+				"agent": schema.BoolAttribute{
```
Comment (Tankilevitch) [inline port/action/schema.go:272]: this breaks API, lets revert this change, and please elaborate on why this is needed?

File: port/action/schema.go [hunk 5]
```diff
@@ -461,7 +460,7 @@ func StringPropertySchema() schema.Attribute {
 			Optional:            true,
 		},
 		"format": schema.StringAttribute{
-			MarkdownDescription: "The format of the string property",
+			MarkdownDescription: "The format of the string property. Common accepted values include `email`,`hostname`,`ipv4`,`ipv6`,`uuid`,`uri`,`date-time`",
```
Comment (Tankilevitch) [inline port/action/schema.go:463]: i am not familiar with those kind of formats, can you refer me to the related docs?

**Review by Tankilevitch — CHANGES_REQUESTED** [review]

File: docs/resources/port_action.md [hunk 6]
```diff
@@ -480,17 +489,17 @@ Optional:
 
 Optional:
 
-- `boolean_items` (Attributes) The items of the array property (see [below for nested schema](#nestedatt--self_service_trigger--user_properties--array_props--boolean_items))
+- `boolean_items` (Attributes) An array of boolean items within the property (see [below for nested schema](#nestedatt--self_service_trigger--user_properties--array_props--boolean_items))
```
Comment (pazhersh) [inline docs/resources/port_action.md:492]: aren't these files auto generated with gen-doc?
if so than these changes will be overriden next time the resources are updated and the docs re-generated

Comment (Tankilevitch) [reply docs/resources/port_action.md:492]: yes, generated by `gen-doc`

File: port/scorecard/schema.go (hunk 1 above)
Comment (kodjomiles) [reply port/scorecard/schema.go:187]: Yeah it will work, it was the only one with the odd structure, the other json types are not like that

File: port/action/schema.go (hunk 2 above)
Comment (kodjomiles) [reply port/action/schema.go:683]: True. the other value is also accepted but let me take out the other one to prevent misleading. since just one is accepted by the validator

File: docs/resources/port_action.md (hunk 6 above)
Comment (pazhersh) [reply docs/resources/port_action.md:492]: Then we should change the `MarkdownDescription` of the properties in the relevant `schema.go` files and rerun gen-doc instead of manually changing the files here

File: port/action/schema.go (hunk 4 above)
Comment (kodjomiles) [reply port/action/schema.go:272]: okay. In the description, the client stated that the agent property is supposed to be a boolean instead of a string attribute. we were described the values of the property as bool and yet it was taken a string property.

[related link](https://registry.terraform.io/providers/port-labs/port-labs/latest/docs/resources/port_action#nestedatt--self_service_trigger--user_properties--array_props--boolean_items)

File: port/action/schema.go (hunk 5 above)
Comment (kodjomiles) [reply port/action/schema.go:463]: I used the JSON Schema Documentation, You can find more information on these formats [here](https://json-schema.org/draft/2020-12/json-schema-validation.html#rfc.section.7.3.1).

File: docs/resources/port_action.md (hunk 6 above)
Comment (kodjomiles) [reply docs/resources/port_action.md:492]: should I leave it be ?

File: port/action/schema.go (hunk 4 above)
Comment (Tankilevitch) [reply port/action/schema.go:272]: this can be also a jq, so this is the reason why its a string and not a boolean, worth elaborating more in the markdown

File: docs/resources/port_action.md (hunk 6 above)
Comment (pazhersh) [reply docs/resources/port_action.md:492]: NVM I see now that the schemas are changed and its not manually edited in the docs 🤦‍♂️

File: port/action/schema.go (hunk 4 above)
Comment (kodjomiles) [reply port/action/schema.go:272]: Makes sense. Let me change it back to String and add more content to the description

File: port/action/schema.go (hunk 5 above)
Comment (pazhersh) [reply port/action/schema.go:463]: The format property in port doesn't actually comply with the official json-schema format; we have our own formats which we've added (such as 'entity')

Please refer to [port's docs](https://docs.getport.io/actions-and-automations/create-self-service-experiences/setup-ui-for-action/user-inputs/#special-string-formats) for the available formats

File: port/action/schema.go (hunk 3 above)
Comment (pazhersh) [reply port/action/schema.go:501]: only `aes256-gsm` is currently supported for property encryption

File: port/action/schema.go [hunk 7]
```diff
@@ -678,7 +677,7 @@ func ObjectPropertySchema() schema.Attribute {
 			},
 		},
 		"encryption": schema.StringAttribute{
-			MarkdownDescription: "The algorithm to encrypt the property with",
+			MarkdownDescription: "The algorithm to encrypt the property with. Accepted value: `aes256-gcm`, `aes128-gcm`",
```
Comment (pazhersh) [inline port/action/schema.go:680]: here too

File: port/action/schema.go [hunk 8]
```diff
@@ -733,24 +732,24 @@ func ArrayPropertySchema() schema.Attribute {
 			},
 		},
 		"string_items": schema.SingleNestedAttribute{
-			MarkdownDescription: "The items of the array property",
+			MarkdownDescription: "An array of string items within the property",
 			Optional:            true,
 			Attributes: map[string]schema.Attribute{
 				"format": schema.StringAttribute{
-					MarkdownDescription: "The format of the items",
+					MarkdownDescription: "The format of the string property. Common accepted values include `email`,`hostname`,`ipv4`,`ipv6`,`uuid`,`uri`,`date-time`",
```
Comment (pazhersh) [inline port/action/schema.go:739]: Same as the other comment about format

**Review by pazhersh — CHANGES_REQUESTED** [review]

File: port/action/schema.go (hunk 5 above)
Comment (kodjomiles) [reply port/action/schema.go:463]: okay got it.. thanks

File: port/action/schema.go (hunk 4 above)
Comment (kodjomiles) [reply port/action/schema.go:272]: @pazhersh what do you think of this 
```
"agent": schema.StringAttribute{
					MarkdownDescription: "Specifies whether to use an agent to invoke the action. This can be a boolean value (`true` or `false`) or a JQ template if dynamic evaluation is needed.",
					Optional:            true,
				}
```

File: port/action/schema.go [hunk 9]
```diff
@@ -461,7 +460,7 @@ func StringPropertySchema() schema.Attribute {
 			Optional:            true,
 		},
 		"format": schema.StringAttribute{
-			MarkdownDescription: "The format of the string property",
+			MarkdownDescription: "The format of the string property. Common accepted values include `date-time`, `url`, `email`, `ipv4`, `ipv6`, `yaml`, `entity`, `user`, `team`, `proto`, `markdown`",
```
Comment (pazhersh) [inline port/action/schema.go:463]:
```suggestion
			MarkdownDescription: "The format of the string property. Accepted values include `date-time`, `url`, `email`, `ipv4`, `ipv6`, `yaml`, `entity`, `user`, `team`, `proto`, `markdown`",
```
since its not really json-schema complient than its not really "common"

File: port/action/schema.go [hunk 10]
```diff
@@ -733,24 +732,24 @@ func ArrayPropertySchema() schema.Attribute {
 			},
 		},
 		"string_items": schema.SingleNestedAttribute{
-			MarkdownDescription: "The items of the array property",
+			MarkdownDescription: "An array of string items within the property",
 			Optional:            true,
 			Attributes: map[string]schema.Attribute{
 				"format": schema.StringAttribute{
-					MarkdownDescription: "The format of the items",
+					MarkdownDescription: "The format of the string property, Common accepted values include `date-time`, `url`, `email`, `ipv4`, `ipv6`, `yaml`, `entity`, `user`, `team`, `proto`, `markdown`",
```
Comment (pazhersh) [inline port/action/schema.go:739]: same as the other comment

**Review by pazhersh — APPROVED** [review]

File: port/action/schema.go [hunk 11]
```diff
@@ -270,9 +270,8 @@ func ActionSchema() map[string]schema.Attribute {
 					Required:            true,
 				},
 				"agent": schema.StringAttribute{
-					MarkdownDescription: "Use the agent to invoke the action",
+					MarkdownDescription: "Specifies whether to use an agent to invoke the action. This can be a boolean value (`'true''` or `'false'`) or a JQ if dynamic evaluation is needed.",
 					Optional:            true,
-					Validators:          StringBooleanOrJQTemplateValidator(),
```
Comment (Tankilevitch) [inline port/action/schema.go:275]: leave it here? why you removed it?

**Review by Tankilevitch — CHANGES_REQUESTED** [review]

File: port/action/schema.go (hunk 11 above)
Comment (kodjomiles) [reply port/action/schema.go:275]: added, thanks
